controllers/marriage-details.js

```diff
@@ -14,11 +14,13 @@ const handleError = (err, next) => {
   return next(err instanceof Error ? err : new Error(err));
 };
 
+const fullDetailsRoleName = 'full-details';
+const showFullDetails = ri => !!(ri && ri.roles && ri.roles.filter(r => r === fullDetailsRoleName).length);
```

Comment:
I like:
* the splitting out of the role name (arguably this could even go into the `config.js` but we can leave that for another day
* the elimination of the reduce step

I don't like:
* the extra function (it's only called once so let's just create the value and avoid the extra layer of abstraction.)
* the error suppression (better to let it throw a wobbler should it get invalid data as that will be much easier to debug)